ui/packages/components/src/Link/Link.tsx

```diff
@@ -1,31 +1,44 @@
+import type { UrlObject } from 'url';
+import type { Route } from 'next';
 import NextLink from 'next/link';
+import { IconArrowRight } from '@inngest/components/icons/ArrowRight';
 import { IconArrowTopRightOnSquare } from '@inngest/components/icons/ArrowTopRightOnSquare';
-import { IconChevron } from '@inngest/components/icons/Chevron';
 import { classNames } from '@inngest/components/utils/classNames';
 
-interface LinkProps {
+type LinkProps<PassedHref extends string> = {
   internalNavigation?: boolean;
   children: React.ReactNode;
   className?: string;
-  href: string;
-}
+  href?: Route<PassedHref> | UrlObject;
```

Comment:
What's the reason for the `Route<PassedHref>` instead of `string`? We end up passing values of type `string` to the `href` property

Reply:
It's required for the <NextLink>. Since the <Link> component is either a Next js link or an anchor, without it we get a `Type error: Type 'string' is not assignable to type 'UrlObject | RouteImpl<string>'.`

Reply:
What does the generic do? Should we use `Route` instead of `Route<PassedHref>` since it looks like we're passing a normal string to it in the `BlankSlate` component?

Reply:
To be honest I don't think we'll need more control over the href prop, so using just the `Route` is good enough for me